File: appPython/services/export_service.py
"""
services/export_service.py - Export deck to Quizlet-compatible format
"""
import os
from models.flashcard import Deck, Flashcard


def export_to_quizlet(
    deck: Deck,
    output_path: str,
    format_type: str = "full",
) -> str:
    """
    Export deck to a Quizlet-importable .txt file.

    format_type options:
    - 'simple'  : Question [TAB] Answer
    - 'full'    : Question >> A | B | C [TAB] Correct answer text
    - 'compact' : Question [A | B] [TAB] Letter(s)
    - 'safe'    : {[(CauHoi)]}Question >> A / B{[(DapAn)]}Answer
                  In Quizlet: Term/Def sep = {[(DapAn)]}, Card sep = {[(CauHoi)]}
    """
    lines = []
    skipped = 0

    for card in deck.cards:
        if not card.question:
            skipped += 1
            continue
        try:
            row = card.to_quizlet_row(format_type)
            lines.append(row)
        except Exception:
            skipped += 1

    if format_type == "safe":
        # For safe mode, cards are already prefixed with {[(CauHoi)]}
        # Join WITHOUT newline — card separator IS {[(CauHoi)]}
        content = "".join(lines)
    else:
        content = "\n".join(lines)

    with open(output_path, "w", encoding="utf-8") as f:
        f.write(content)

    return f"Exported {len(lines)} cards ({skipped} skipped) to:\n{output_path}"


def get_quizlet_preview(deck: Deck, format_type: str = "full", max_rows: int = 5) -> str:
    """Return a preview of the first few rows for display."""
    preview_lines = []
    for card in deck.cards[:max_rows]:
        if card.question:
            preview_lines.append(card.to_quizlet_row(format_type))
    # For display, always join with double newline for readability
    return "\n\n".join(preview_lines)

File: appDotNet/FlashcardAI/Services/ExportService.cs
using System.Collections.Generic;
using System.IO;
using System.Linq;
using FlashcardAI.Models;

namespace FlashcardAI.Services;

/// <summary>
/// Export deck to Quizlet-compatible formats.
/// </summary>
public static class ExportService
{
    public static string ExportToQuizlet(Deck deck, string outputPath, string formatType = "full")
    {
        var lines = new List<string>();
        int skipped = 0;

        foreach (var card in deck.Cards)
        {
            if (string.IsNullOrWhiteSpace(card.Question)) { skipped++; continue; }
            try { lines.Add(card.ToQuizletRow(formatType)); }
            catch { skipped++; }
        }

        string content = formatType == "safe"
            ? string.Concat(lines)
            : string.Join("\n", lines);

        File.WriteAllText(outputPath, content, System.Text.Encoding.UTF8);
        return $"Exported {lines.Count} cards ({skipped} skipped) to:\n{outputPath}";
    }

    public static string GetQuizletPreview(Deck deck, string formatType = "full", int maxRows = 5)
    {
        var lines = deck.Cards.Take(maxRows)
            .Where(c => !string.IsNullOrWhiteSpace(c.Question))
            .Select(c => c.ToQuizletRow(formatType))
            .ToList();
        return string.Join("\n\n", lines);
    }
}
